pipeline/backend/kubernetes/kubernetes.go

```diff
@@ -266,8 +266,18 @@ func (e *kube) WaitStep(ctx context.Context, step *types.Step, taskUUID string)
 		return nil, fmt.Errorf("could not pull image for pod %s", pod.Name)
 	}
 
+	if len(pod.Status.ContainerStatuses) == 0 {
+		return nil, fmt.Errorf("No container statuses found for pod %s", pod.Name)
+	}
+
+	cs := pod.Status.ContainerStatuses[0]
+
+	if cs.State.Terminated == nil {
+		return nil, fmt.Errorf("No terminated state found for container %s/%s", pod.Name, cs.Name)
```

Comment:
A bit, yeah

I could add something like this

```
	if cs.State.Terminated == nil {
		err := fmt.Errorf("no terminated state found for container %s/%s", pod.Name, cs.Name)
		log.Error().Str("taskUUID", taskUUID).Interface("state", cs.State).Msg(err.Error())
		return nil, err
	}
```

Note that when I caught this issue the node had been killed mid-execution, and there were kubernetes events for terminating the pod correctly. So I guess it just never got "assigned" a termination status, but the pod was indeed terminated. In other words, whatever status it reported instead of terminated was certainly "bonkers".